packages/nextjs/src/server/authMiddleware.ts

```diff
@@ -132,6 +132,10 @@ export interface AuthMiddleware {
   (params?: AuthMiddlewareParams): NextMiddleware;
 }
 
+/**
+ * @deprecated Use `clerkMiddleware` instead.
+ * Migration guide: https://clerk.com/docs/upgrade-guides/v5-introduction
+ */
```

Comment:
🔧  let's add the deprecation when `clerkMiddleware` is added and also add a deprecation warning to the `authMiddleware` using the `@clerk/shared/deprecated` in v5 version

Reply:
I don't think we need to wait, `clerkMiddleware` will be released in v5 so both this warning and the new middleware will be released at the same time :)

In any case, clerkMiddleware will be introduced shortly

Reply:
I think it's more related to add the deprecation in the same PR or in a PR later than when that change is applied. That's why. But I'm fine with it.

Reply:
Ah, okay just now got what you mean. Yes that makes more sense :)